feat: Add terms-and-conditions sample app for CI testing

swan-amazon opened the pull request:
Add a new sample application that demonstrates and tests the terms and conditions feature. This app will be integrated into the CI pipeline to prevent regressions and ensure the feature continues to work as expected.

The sample app serves as both a reference implementation and an automated test case for the terms and conditions functionality.

### Testing

Used the chip-tool to commission chip-terms-and-conditions-app. Confirmed that the commissioning succeeds when terms are sent and fails when terms are not sent.

```
chip-terms-and-conditions-app --version 0 --custom-flow 2 --capabilities 6 --discriminator 3840 --passcode 20202021 --tc-min-required-version 1 --tc-required-acknowledgements 1 --KVS /tmp/chip_kvs.bin --trace_file /tmp/chip_trace.log --trace_log 1 --trace_decode 1

chip-tool pairing code 1 749701123365521327694 --trace_decode 1 --require-tc-acknowledgements 1 --tc-acknowledgements 1 --tc-acknowledgements-version 1 --bypass-attestation-verifier true
```

swan-amazon committed:
feat: Add terms-and-conditions sample app for CI testing

Add a new sample application that demonstrates and tests the terms and
conditions feature. This app will be integrated into the CI pipeline to
prevent regressions and ensure the feature continues to work as
expected.

The sample app serves as both a reference implementation and an
automated test case for the terms and conditions functionality.

```bash
./scripts/build/build_examples.py --target linux-x64-terms-and-conditions build
```

bzbarsky-apple commented on examples/terms-and-conditions-app/linux/main.cpp:
```diff
@@ -0,0 +1,86 @@
+{
+    const app::TermsAndConditions termsAndConditions = app::TermsAndConditions(sTcRequiredAcknowledgements, sTcMinRequiredVersion);
+
+    const Optional<app::TermsAndConditions> requiredTermsAndConditions = Optional<app::TermsAndConditions>(termsAndConditions);
```
```suggestion
    auto requiredTermsAndConditions = MakeOptional(termsAndConditions);
```

Of course you could also pass `MakeOptional(termsAndConditions)` directly to the `Init()` below.

swan-amazon replied:
Thanks. I was attempting to avoid line-breaks. Updated per suggestion.

bzbarsky-apple commented on examples/terms-and-conditions-app/linux/main.cpp:
```diff
@@ -0,0 +1,86 @@
+
+    PersistentStorageDelegate & persistentStorageDelegate = Server::GetInstance().GetPersistentStorage();
+
+    chip::app::TermsAndConditionsManager::GetInstance()->Init(&persistentStorageDelegate, requiredTermsAndConditions);
```
```suggestion
    app::TermsAndConditionsManager::GetInstance()->Init(&persistentStorageDelegate, requiredTermsAndConditions);
```

swan-amazon replied:
Updated

bzbarsky-apple commented on examples/terms-and-conditions-app/terms-and-conditions-common/terms-and-conditions-app.matter:
```diff
@@ -0,0 +1,1796 @@
+    ram      attribute clusterRevision default = 3;
+  }
+
+  server cluster LocalizationConfiguration {
```
Does this app actually implement LocalizationConfiguration in any useful way?

swan-amazon replied:
Thanks! updated in new PR rev.

bzbarsky-apple commented on examples/terms-and-conditions-app/terms-and-conditions-common/terms-and-conditions-app.matter:
```diff
@@ -0,0 +1,1796 @@
+    callback attribute clusterRevision;
+  }
+
+  server cluster ScenesManagement {
```
This is provisional, right?  I suggest not enabling it...

swan-amazon replied:
Thanks! updated in new PR rev.

swan-amazon committed:
Update cluster configurations for app testing

- Enable switch cluster for basic app control testing
- Enable network commissioning cluster for in-app commission testing
- Disable provisional clusters as they are not required
- Disable other non-required clusters to streamline testing setup

This change focuses the cluster configuration on essential components
needed for basic app control and commissioning testing scenarios.

bzbarsky-apple commented on examples/terms-and-conditions-app/terms-and-conditions-common/terms-and-conditions-app.matter:
```diff
@@ -0,0 +1,1658 @@
+    callback attribute clusterRevision;
+  }
+
+  server cluster Switch {
```
This cluster is not listed for this device type in the spec, right?

swan-amazon replied:
For demo integration with Alexa app I want one ux control. The demo I gave at the MM had this cluster enabled for that purpose.

swan-amazon replied:
For demo integration with Alexa app I want one ux control. The demo I gave at the MM had this cluster enabled for that purpose.

bzbarsky-apple replied:
That doesn't actually address the "this is not really spec-compliant" question.  If you are a switch, you should have the right device type to be a switch.

swan-amazon replied:
I updated the zap and matter file to attempt to comply with the "On/Off Light" device type. Please take a look and let me know if it looks correct. Using different versions of the zap tool, I get warning messages related to compliance due to the Provisional attributes.

swan-amazon committed:
Merge remote-tracking branch 'origin/master' into pr/swan-amazon/36950

swan-amazon committed:
Merge remote-tracking branch 'origin/master' into pr/swan-amazon/36950

swan-amazon committed:
refactor: Remove unused attributes from Switch cluster

Remove unused attributes from the Switch cluster to comply with the "On/Off Light"
device type specification as defined in the Matter device library v1.4. This change
removes several server-side attributes including ServerList, ClientList, PartsList,
and various internal lists that are not required for the basic On/Off Light
functionality.

swan-amazon committed:
feat: Add Descriptor cluster and enable attribute reporting

Add the mandatory Descriptor cluster to endpoint 1 and enable attribute reporting
for the device. Changes include:
- Add Descriptor cluster with all required attributes (deviceTypeList, serverList, etc.)
- Enable reportable flag for multiple attributes with min/max reporting intervals
- Configure all Descriptor cluster attributes as external callbacks

This change ensures compliance with the Matter specification for the On/Off Light
device type.
